### PR TITLE
RDoc-884 what is collection

### PR DESCRIPTION
**Pull request by talweiss1982**

### CODE REVIEW

File: Documentation/4.0/Raven.Documentation.Pages/client-api/faq/what-is-a-collection.markdown [hunk 1]
```diff
@@ -0,0 +1,28 @@
+﻿#What is a collection?
+
+A collection in RavenDB is a set of documents with the same tag. The tag is defined in `@collection` metadata property which is filled in by the client based on the type of 
+an entity object that you store (the function responsible for tagging documents can by overwritten by using [customizations](../../client-api/configuration/conventions/identifier-generation/global#findtypetagname-and-finddynamictagname)). Note that documents that are in
```
Comment (arekpalinski) [inline Documentation/4.0/Raven.Documentation.Pages/client-api/faq/what-is-a-collection.markdown]: Actually we no longer use "tag" word regarding collection. In 3.x we had `Tag` in `Raven/DocumentsByEntityName` that's why we refereed to it when talking about collection. Maybe just explain that it's "a set of documents having the same `@collection` medatata property"?

Yes, typically is filled based on entity type from the client. However note that if you create a document using studio with id like `users|`, it will get `"@collection": "Users"` as well.

The link `[customizations]` won't be valid as `FindTypeTagName` and `FindDynamicTagName` got renamed to `FindCollectionName` and `FindCollectionNameForDynamic`. You can leave it just empty for now

File: Documentation/4.0/Raven.Documentation.Pages/client-api/faq/what-is-a-collection.markdown [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+has a great meaning for three RavenDB features: [the studio](../../studio/overview/documents/documents-view), [the indexes](../../indexes/what-are-indexes) and [the document key generation](../../client-api/document-identifiers/working-with-document-ids) on the client side.
+
+##Collections usage
+
+###Studio
+
+Probably the first time you encounter the collection will be while accessing the studio. Then you will see that, for example, the `Order` entity that you have just stored is visible under 
+`Orders` collection (by default the client pluralizes the collection name based on the type name). But how does it happen that the virtual concept of the collections is
+visualized in the studio. The answer is that each RavenDB database has a built-in index per collection, which is used internally to query the database and retrieve
+only  documents from the specified collection. This way the studio can group the documents into the collections.
```
Comment (arekpalinski) [inline Documentation/4.0/Raven.Documentation.Pages/client-api/faq/what-is-a-collection.markdown]: `a built-in index per collection` - maybe mention it's storage index to avoid confusion

File: Documentation/4.0/Raven.Documentation.Pages/client-api/faq/what-is-a-collection.markdown [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+
+###Indexing
+
+Another usage of collections is filtering documents during indexing process. When you create an index you determine what collection does it involve. During indexing process only the documents
```
Comment (arekpalinski) [inline Documentation/4.0/Raven.Documentation.Pages/client-api/faq/what-is-a-collection.markdown]: Actually we don't need to filter documents out during indexing since we iterate directly on docs from collection - there is no need to filter them out as we store collection documents in separate tables

File: Documentation/4.0/Raven.Documentation.Pages/client-api/faq/what-is-a-collection.markdown [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+that belong this collection are indexed, the rest is filtered out.
+
+###Document keys
+
+The default convention is that documents have the identifiers like `orders/1` where `orders` is just a collection name and `1` is the identity value.
```
Comment (arekpalinski) [inline Documentation/4.0/Raven.Documentation.Pages/client-api/faq/what-is-a-collection.markdown]: now by default we create ids like `orders/1-A`

File: Documentation/4.0/Raven.Documentation.Pages/studio/overview/overview.markdown [hunk 5]
```diff
@@ -0,0 +1,41 @@
+﻿## Studio Overview
```
Comment (arekpalinski) [inline Documentation/4.0/Raven.Documentation.Pages/studio/overview/overview.markdown]: it doesn't seem to be part of this PR intentionally

File: Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/DocumentIdentifiers/HiLoAlgorithm.cs [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+{
+	public class HiLoAlgorithm
+	{
+		public HiLoAlgorithm()
+		{
+			var store = new DocumentStore();
+
+			#region replication_hilo
+
+			store.DatabaseCommands.Put("Raven/ServerPrefixForHilo", null,
```
Comment (arekpalinski) [inline Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/DocumentIdentifiers/HiLoAlgorithm.cs]: invalid code, it doesn't seem to be part of this PR intentionally

File: Documentation/4.0/Raven.Documentation.Pages/client-api/faq/what-is-a-collection.markdown (hunk 3 above)
Comment (talweiss1982) [reply Documentation/4.0/Raven.Documentation.Pages/client-api/faq/what-is-a-collection.markdown]: how do we iterate the documents for multi map indexes?

Comment (arekpalinski) [reply Documentation/4.0/Raven.Documentation.Pages/client-api/faq/what-is-a-collection.markdown]: each collection is iterated independently

File: Documentation/4.0/Raven.Documentation.Pages/studio/overview/overview.markdown (hunk 5 above)
Comment (talweiss1982) [reply Documentation/4.0/Raven.Documentation.Pages/studio/overview/overview.markdown]: yes i decided we should have a separate issue for the stubs i'll open those issues

File: Documentation/4.0/Raven.Documentation.Pages/client-api/faq/what-is-a-collection.markdown (hunk 4 above)
Comment (talweiss1982) [reply Documentation/4.0/Raven.Documentation.Pages/client-api/faq/what-is-a-collection.markdown]: yes, i'll open an issue to elaborate more on document identifiers.

File: Documentation/4.0/Raven.Documentation.Pages/studio/overview/overview.markdown (hunk 5 above)
Comment (talweiss1982) [reply Documentation/4.0/Raven.Documentation.Pages/studio/overview/overview.markdown]: okay, i'll modify the documentation.
I thought that we index based on Etag order.
So we do a batch of one collection and a batch of the other collection interleaved ?

Comment (arekpalinski) [reply Documentation/4.0/Raven.Documentation.Pages/studio/overview/overview.markdown]: yes, we still index based on etag order. but we keep the last indexed etag per collection. 

We index collections one by one in the same transaction

Comment (talweiss1982) [reply Documentation/4.0/Raven.Documentation.Pages/studio/overview/overview.markdown]: okay checkout my new text

File: Documentation/4.0/Raven.Documentation.Pages/client-api/faq/what-is-a-collection.markdown [hunk 7]
```diff
@@ -1,8 +1,10 @@
 ﻿#What is a collection?
 
-A collection in RavenDB is a set of documents with the same tag. The tag is defined in `@collection` metadata property which is filled in by the client based on the type of 
-an entity object that you store (the function responsible for tagging documents can by overwritten by using [customizations](../../client-api/configuration/conventions/identifier-generation/global#findtypetagname-and-finddynamictagname)). Note that documents that are in 
-the same collection can have a completely different structure, which is fine because RavenDB is schema-less.
+A collection in RavenDB is a set of documents with the same `@collection` metadata property which is filled in by the client based on the type of 
+an entity object that you store (the function responsible for tagging documents can by overwritten by using [customizations](../../client-api/configuration/conventions/identifier-generation/global#findtypetagname-and-finddynamictagname)). 
+Also if the documents are inserted through the studio a `@collection` metadata will be generated for them, e.g `users|`/`users/`/`users/17` will have `@collection:users`.
```
Comment (arekpalinski) [inline Documentation/4.0/Raven.Documentation.Pages/client-api/faq/what-is-a-collection.markdown]: will have `"@collection": "Users"` - upper cased

File: Documentation/4.0/Raven.Documentation.Pages/client-api/faq/what-is-a-collection.markdown [hunk 8]
```diff
@@ -21,8 +21,8 @@ only  documents from the specified collection. This way the studio can group the
 
 ###Indexing
 
-Another usage of collections is filtering documents during indexing process. When you create an index you determine what collection does it involve. During indexing process only the documents
-that belong this collection are indexed, the rest is filtered out.
+Each Ravendb index is built against a collection (or collections when using [multi map index](../../indexes/multi-map-indexes)) we use a storage index to iterate documents of the same collection in the indexing process from this
+you may derive that a collection is a set of documents that have sence been indexed together.
```
Comment (arekpalinski) [inline Documentation/4.0/Raven.Documentation.Pages/client-api/faq/what-is-a-collection.markdown]: sence? 

"a set of documents that have sence been indexed together." - what if the index definition does some filtering on its own e.g.

`from u in docs.Users where u.Age > 30 select { u.Name }`

I would skip this part of sentence completely

File: Documentation/4.0/Raven.Documentation.Pages/client-api/faq/what-is-a-collection.markdown [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+
+Probably the first time you encounter the collection will be while accessing the studio. Then you will see that, for example, the `Order` entity that you have just stored is visible under 
+`Orders` collection (by default the client pluralizes the collection name based on the type name). But how does it happen that the virtual concept of the collections is
+visualized in the studio. The answer is that each RavenDB database has a storage index per collection, which is used internally to query the database and retrieve
+only  documents from the specified collection. This way the studio can group the documents into the collections.
+
+
+###Indexing
+
+Each Ravendb index is built against a collection (or collections when using [multi map index](../../indexes/multi-map-indexes)) we use a storage index to iterate documents of the same collection in the indexing process.
```
Comment (ayende) [inline Documentation/4.0/Raven.Documentation.Pages/client-api/faq/what-is-a-collection.markdown]: Storage index is not going to be familiar to the user, better "iterate over the collection's documents"

File: Documentation/4.0/Raven.Documentation.Pages/studio/overview/overview.markdown [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+---
+
+{PANEL}
+RavenDB comes with a Management Studio that can be accessed by any modern browser using the server URL     
+(e.g. http://localhost:8080/ - when your server runs at port 8080).
```
Comment (ayende) [inline Documentation/4.0/Raven.Documentation.Pages/studio/overview/overview.markdown]: With the default setup wizard, maybe direct them to look at the console output for the actual value

Comment (talweiss1982) [reply Documentation/4.0/Raven.Documentation.Pages/studio/overview/overview.markdown]: this is just me moving the layout of the overview to match previous version, there were already other PR conflicting with mine, i have selected theirs and rebased.

**Comment by talweiss1982** [comment]
done

**Comment by ppekrol** [comment]
@talweiss1982 Arek will review this when he will be back from holidays
